### DISCUSSION TITLE
Node.js | Follow dist tag automatically

### DISCUSSION THREAD

**Discussion by SchulteDev** [renovatebot/renovate#29302] [Suggest an Idea]
# Context

## 1. Node.js npm-dist-tag

In Node.js, there is the option to use a [npm-dist-tag](https://docs.npmjs.com/cli/v10/commands/npm-dist-tag) in the package.json  like this:

```json
"devDependencies": {
  "nx": "canary"
}
```

## 2. Renovate followTag

Renovate has the [followTag-option](https://docs.renovatebot.com/configuration-options/#followtag). When used, Renovate will update a dependency based on the change to a *tag*.

```json5
// Renovate config:
{
  "packageRules": [
    {
      "matchPackageNames": ["nx"],
      "followTag": "canary"
    }
  ]
}

// package.json
"devDependencies": {
  "nx": "19.2.0-canary.20240528-7f11a1d" // this will get updated by Renovate when "canary" tag at https://www.npmjs.com/package/nx changes
}
```

# Suggestion

When there is a dist-tag defined in the package.json like this

```json
"devDependencies": {
  "nx": "canary"
}
```

Renovate updates the package automatically when the tag is being changed in the packages' registry.

- followTag option would not have to be set
- lock-file (NPM, PNPM, ...) would be updated, but package.json would not

## Example

I have created https://github.com/SchulteMarkus/renovate-npm-dist-tag for a tiny demo. The included dependency `nx=canary` will never be updated by Renovate.

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/29303
